modules/library/jdbc/src/main/java/org/geotools/jdbc/KeysFetcher.java

```diff
@@ -420,6 +423,27 @@ public Object getNext(Connection cx) {
         }
     }
 
+    private static class FromUuid extends KeyFetcher {
```

Comment:
New code, needs a test. Suggest adding it in the test suite of a database supporting uuids, like postgresql.

Reply:
Will do.  Would this be a good place to put the unit test:  modules\plugin\jdbc\jdbc-postgis\src\test\java\org\geotools\data\postgis\PostGISUuidTestSetup.java ?

Reply:
Please provide some guidance here.  Is it acceptable to test this at the geoserver level, seeing as that is where the issue was reported?  Then I would create a unit test with insert feature by WFS-T.  Otherwise, I don't know how to test it in geotools.  Do I need to set up mock connections, stores, featuresource, keys, etc.  There are no previous unit tests to copy from.

Reply:
The test should be done in GeoTools. There are plenty of JUnit tests talking to a database in the gt-jdbc-postgresql, you probably need to work either on PostgisPrimaryKeyFinderOnlineTest, or PostgisPrimaryKeyOnlineTest, or PostGISUuidOnlineTest, extending the associated test setup, and probably adding a new test method in the postgis specific subclasses (e.g., PostGISUuidOnlineTest) rather than the base class shared across databases (e.g., JDBCUuidTestSetup).

When you try to run it, it will create a $HOME/.geotools/postgis.properties.example file and fail, rename the created file it into postgis.properties and fix the connection properties to point to a database that one can write onto, run it again, make it pass.

Reply:
Thank you @aaime I had subsequently figured out that much, and just got the online tests to run, but appreciate guidance on where to put it.

I assume it's too late to go into the upcoming geotools release, being 21.5, so I will get back to it in the next week or so, hopefully.

Reply:
Yes GeoTools 21.5 is already cut... and it's the last one of the 21.x series. But 22.x is alive for another 6 months and will have a new release in April (in March we release 23.0).